src/components/text-pane/text-pane.tsx

```diff
@@ -22,14 +26,33 @@ export const TextPane = observer(function TextPane() {
     );
   }, []);
 
+  // Update the title when the target dataset, title, or attribute changes
+  useEffect(() => {
+    return reaction(
+      () => {
+        return [
+          textStore.titleDataset, targetDatasetStore.targetDatasetInfo.title, targetStore.targetAttributeName,
+          testingStore.testingDatasetInfo.title, testingStore.testingAttributeName
+        ];
+      },
+      () => {
+        const dataset = textStore.titleDataset === "target"
+          ? targetDatasetStore.targetDatasetInfo.title : testingStore.testingDatasetInfo.title;
+        const attribute = textStore.titleDataset === "target"
+          ? targetStore.targetAttributeName : testingStore.testingAttributeName;
+        textStore.updateTitle(dataset, attribute);
+      }
+    );
+  }, []);
+
```

Comment:
You don't need to change it, but I think you could make `textStore.textComponentTile` a view which computes this, so then it wouldn't be necessary to add an explicit reaction like this. Or if having the textStore depend on the teh targetDatasetStore, targetStore, and testingStore is a problem then you could make a new store.

Reply:
Big improvement, thanks for the suggestion 👍